renter/hostdb/update.go

```diff
@@ -0,0 +1,107 @@
+package hostdb
+
+import (
+	"math"
+
+	"go.sia.tech/core/types"
+)
+
+func updateHostHistoricInteractions(host *Host, index types.ChainIndex) {
```

Comment:
I question whether we should use a `ChainIndex` here. It seems like what we really care about is *time*, and we're just using the block height as a stand-in for that.